lib/signs/bus.ex

```diff
@@ -0,0 +1,106 @@
+defmodule Signs.Bus do
+  use GenServer
+  require Logger
+
+  def start_link(sign, opts \\ []) do
+    %{
+      "type" => "bus",
+      "id" => id,
+      "pa_ess_loc" => pa_ess_loc,
+      "text_zone" => text_zone,
+      "audio_zones" => audio_zones,
+      "sources" => sources
```

Comment:
This pattern match does a good job of getting these values and checking for their existence, since the pattern match will throw an exception if any of these are missing in `sign`, but it might be even better to use `Map.fetch!` to get some of these like we do in `realtime.ex`. The advantage is a more specific exception if/when a key is missing from the map.

Reply:
The match error would look something like `(MatchError) no match of right hand side value: %{a: 1, b: 2, c: 3}` where the right hand side value would be the value of `sign`. You'd have to take a look at the pattern match in the code and compare it with the logged state of `sign` to figure out which key it was that you're missing. Trivial when the map is small, but less so as it gets bigger.

The error from `Map.fetch!` would look something like `(KeyError) key :d not found in: %{a: 1, b: 2, c: 3} (stdlib 3.17) :maps.get(:d, %{a: 1, b: 2, c: 3})`  so we'd immediately know what key was missing from the config.

And yeah, I do agree that the pattern matching is a nice way to convey the structure of the data. But since we'll be including pretty much everything in `sign` in our struct definition, aren't we already pretty clearly conveying the shape of the data?